beacon-chain/rpc/beacon/blocks.go

```diff
@@ -191,6 +193,28 @@ func (bs *Server) StreamBlocks(_ *ptypes.Empty, stream ethpb.BeaconChain_StreamB
 					// One nil block shouldn't stop the stream.
 					continue
 				}
+				genesisRoot, err := bs.BeaconDB.GenesisBlockRoot(stream.Context())
+				if err != nil {
+					return status.Errorf(codes.Internal, "Could not genesis block root: %v", err)
+				}
+				beaconState, err := bs.HeadFetcher.HeadState(stream.Context())
+				if err != nil {
+					return err
+				}
+				proposer, err := beaconState.ValidatorAtIndex(data.SignedBlock.Block.ProposerIndex)
+				if err != nil {
+					return err
+				}
+
+				// Verify proposer signature before sending to stream.
+				currentEpoch := helpers.SlotToEpoch(beaconState.Slot())
+				domain, err := helpers.Domain(beaconState.Fork(), currentEpoch, params.BeaconConfig().DomainBeaconProposer, genesisRoot[:])
```

Comment:
the spec says genesis validator root, not genesis block root. You can get it from the state

Reply:
thanks, will do

Reply:
changed